test/d2l-filter-dropdown/d2l-filter-dropdown.js

```diff
@@ -79,289 +79,317 @@ import { afterNextRender } from '@polymer/polymer/lib/utils/render-status.js';
-				for (var i = 0; i < pages.length; i++) {
-					var opts = _getOptions(pages[i]);
-					assert.equal(pages[i].options.length, opts.length);
-					for (var j = 0; j < opts.length; j++) {
-						assert.equal (opts[j].selected, expected[i].options[j].selected);
-						assert.equal(opts[j].text, expected[i].options[j].title);
+			test('updating filter option updates filters', function() {
+				const update = 2;
+				_setSelectedOptions([0, 4, 5]);
+				const expected = _getExpectedAndImport(filter);
+				filter.updateFilterOptionSelectedStatus(options[update].cat, options[update].key, true);
+				_updateExpectedOption(update, true, '', expected);
+				assert.deepEqual(expected, filter._filters);
+			});
+			test('updating filter option updates display', function(done) {
+				const update = 2;
+				const selected = [0, 4, 5];
+				_setSelectedOptions(selected);
+				const expected = _getExpectedAndImport(filter);
+				filter.updateFilterOptionSelectedStatus(options[update].cat, options[update].key, true);
+				_updateExpectedOption(update, true, '', expected);
+				window.requestAnimationFrame(function() {
+					var pages = _getPages();
+					for (var i = 0; i < pages.length; i++) {
+						var opts = _getOptions(pages[i]);
+						assert.equal(pages[i].options.length, opts.length);
+						for (var j = 0; j < opts.length; j++) {
+							assert.equal (opts[j].selected, expected[i].options[j].selected);
+						}
 					}
-				}
-				done();
+					done();
+				});
+			});
+			test('adding duplicate filter option updates filters', function() {
+				const update = 2;
+				const newTitle = 'newTitle';
+				_setSelectedOptions([0, 4, 5]);
+				const expected = _getExpectedAndImport(filter);
+				filter.addFilterOption(options[update].cat, options[update].key, newTitle, true);
+				_updateExpectedOption(update, true, newTitle, expected);
+				assert.deepEqual(expected, filter._filters);
+			});
+			test('adding duplicate filter option updates display', function(done) {
+				const update = 2;
+				const newTitle = 'newTitle';
+				const selected = [0, 4, 5];
+				_setSelectedOptions(selected);
+				const expected = _getExpectedAndImport(filter);
+				filter.addFilterOption(options[update].cat, options[update].key, 'newTitle', true);
+				_updateExpectedOption(update, true, newTitle, expected);
+				window.requestAnimationFrame(function() {
+					var pages = _getPages();
+					for (var i = 0; i < pages.length; i++) {
+						var opts = _getOptions(pages[i]);
+						assert.equal(pages[i].options.length, opts.length);
+						for (var j = 0; j < opts.length; j++) {
+							assert.equal (opts[j].selected, expected[i].options[j].selected);
+							assert.equal(opts[j].text, expected[i].options[j].title);
+						}
+					}
+					done();
+				});
 			});
 		});
 	});
 	suite('d2l-filter-dropdown-page', function() {
 		setup(function() {
 			pageFixture = fixture('page');
 		});
-		test('contains returns true when substring exists', function() {
```

Comment:
Since I deleted `_caseInsensitiveContainsSubstring` I also deleted the 4 associated tests